relay/api/schemas.py

```diff
@@ -68,6 +73,26 @@ class UserExchangeEventSchema(ExchangeEventSchema):
     direction = fields.Str()
 
 
+class AnyEventSchema(OneOfSchema):
+    type_schemas = {
+        'CurrencyNetworkEvent': UserCurrencyNetworkEventSchema,
+        'UnwEthEvent': UserTokenEventSchema,
+        'ExchangeEvent': ExchangeEventSchema,
+    }
+
+    type_field = '__class__'
+
+    def get_obj_type(self, obj):
```

Comment:
I think this is the default implementation, so you dont have to override this method I guess?

Reply:
I'm using the exact logic that has been in use before. It differs from the default implementation because it's using isinstance checks and will return the same object type for all subclasses.

Reply:
what I mean was that AFAIK this is the same what the default logic of get_object_type is doing so I would just remove the function.

Reply:
what I tried to eplain is that this is not the case. It differs from the default implementation.

Reply:
Okay maybe I am wrong. It looks like you just return the class name and on the webseite they state this is the default implementation
> By default get_obj_type() returns obj.__class__.__name__, so you can just reuse that to save some typing:

Reply:
> Okay maybe I am wrong. It looks like you just return the class name and on the webseite they state this is the default implementation

The result differs for a BalanceUpdateEvent, which is a subclass of CurrencyNetworkEvent.

This doesn't mean that we couldn't or shouldn't use the default implementation, just that it differs.

Maybe dispatching based on the type field would be better. I have to take a second look at it. Ideally we would be able to load that data again with marshmallow. This is currently not possible AFAICT.

Reply:
opened in an issue here: https://github.com/trustlines-network/relay/issues/209